### ISSUE TITLE
Ternary operator in require

### ISSUE THREAD

**Issue by zjjw** [webpack/webpack#627]
I have encountered a problem bundling the following code:

```js
function multiRequire(module1, module2) {
    return require(util.isNode() ? module1 : module2);
}
```
(pulled from https://github.com/aws/aws-sdk-js/blob/master/lib/util.js#L43)

I have attempted to disable the handling of dynamic requires using the following config.

```js
{
  module: {
    // Disable handling of unknown requires
    unknownContextRegExp: /$^/,
    unknownContextCritical: false,

    // Disable handling of requires with a single expression
    exprContextRegExp: /$^/,
    exprContextCritical: false,
  }
}
```

The error thrown is "`Cannot convert options with mixed known and unknown stuff`" (https://github.com/webpack/webpack/blob/master/lib/dependencies/CommonJsRequireDependencyParserPlugin.js#L44).

The problem is (I think) that webpack does not understand a require that has been called with a conditional. The error message is a bit vague though. (What is "`stuff`"? What are "`options`"?).

Additionally, if the ternary is converted to an if/else statement then webpack does not throw an error. Short of rewriting this function, how can I webpack this code?

**Comment by zjjw** [comment]
Any thoughts on this?

**Comment by 1000hz** [comment]
Running into this exact issue, also trying to load aws-sdk via webpack.

**Comment by xdissent** [comment]
:+1: also using aws-sdk

**Comment by ryanmt** [comment]
:+1:

**Comment by joearasin** [comment]
Is there any reason not to remove multiRequire entirely? Unless I'm missing something, it looks like dead code.

**Comment by anthonator** [comment]
:+1:

**Comment by pilwon** [comment]
:+1: x 100

**Comment by joearasin** [comment]
zjjw -- presumably, the first is treated by requiring both modules, then deciding at runtime which one to execute. With regards to the second, it's unclear what the 'require' is supposed to do.

**Comment by Dashed** [comment]
Wow. Awesome to see this is now possible!

**Comment by sokra** [comment]
It doesn't crash anymore. `?:` in require with expressions now behaves like any other expression in require.

**Comment by zjjw** [comment]
:sparkles: ty

**Comment by anthonator** [comment]
Was this released in 1.9.4? I'm still having this issue in the `aws-sdk` after upgrading.

**Comment by anthonator** [comment]
I was using `gulp-webpack` which hasn't transitioned to `1.9` yet. Fixed now.
